### PR TITLE
Added a better saving method

### PR DESCRIPTION
**Pull request by r1walz**
Save better quality PNGs using output stream. Previously we were using insertImage to save the image to the gallery.

### CODE REVIEW

File: app/src/main/java/com/sdsmdg/kd/trianglifyexample/MainActivity.java [hunk 1]
```diff
@@ -225,7 +234,10 @@ public int pxToDp(int px) {
     public boolean onOptionsItemSelected(MenuItem item) {
         switch (item.getItemId()) {
             case R.id.action_save:
-                exportImage();
+                try {
+                    exportImage();
+                } catch (IOException ignored) {
+                }
```
Comment (suyashmahar) [inline app/src/main/java/com/sdsmdg/kd/trianglifyexample/MainActivity.java]: Catch this exception with an error message, toast notification would do, without one user won't know what went wrong.

File: app/src/main/java/com/sdsmdg/kd/trianglifyexample/MainActivity.java [hunk 2]
```diff
@@ -328,7 +344,11 @@ public void onRequestPermissionsResult(int requestCode,
             case PERMISSION_CODE:
                 if (grantResults.length > 0
                         && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
-                    exportImage();
+                    try {
+                        exportImage();
+                    } catch (IOException ignored) {
+
```
Comment (suyashmahar) [inline app/src/main/java/com/sdsmdg/kd/trianglifyexample/MainActivity.java]: This one too

**Review by suyashmahar — CHANGES_REQUESTED** [review]
Nice work @r1walz, just fix these and we'll be good to merge this.

**Comment by suyashmahar** [comment]
@r1walz One last thing, please upload an apk here when you make these changes.

**Comment by r1walz** [comment]
https://goo.gl/AjBeMV
Added a toast message to help the user identify the error.

File: app/src/main/java/com/sdsmdg/kd/trianglifyexample/MainActivity.java (hunk 1 above)
Comment (r1walz) [reply app/src/main/java/com/sdsmdg/kd/trianglifyexample/MainActivity.java]: should I just Toast out error.getMessage()?

**Review by suyashmahar — APPROVED** [review]

**Pull request merged**